include/fcl/common/visibility.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#ifndef _FCL_VISIBLE_HH_
+#define _FCL_VISIBLE_HH_
+
+/** \def FCL_VISIBLE
+ * Use to represent "symbol visible" if supported
+ */
+
+/** \def FCL_HIDDEN
+ * Use to represent "symbol hidden" if supported
+ */
+
+#if defined _WIN32 || defined __CYGWIN__
+  #ifdef BUILDING_DLL
```

Comment:
In Gazebo, the usage of the same `BUILDING_DLL` macro for different libraries created several problems (see https://bitbucket.org/osrf/gazebo/issues/2262/symbol-visibility-issues-in-windows). I would recommend to use a fcl-specific macro and set the [`DEFINE_SYMBOL`](https://cmake.org/cmake/help/v3.5/prop_tgt/DEFINE_SYMBOL.html) property of the library appropriately.

Reply:
The header in this form is breaking the static builds in MSVC, that is the default behaviour when building the library in MSVC.

Reply:
Oh sorry forgot about this issue, thanks for the reminder Silvio. Could please give a look at this patch https://github.com/j-rivero/fcl/commit/e509a3d1659591bb4bacc98638ff74e505c5fee7 and let me know if it is correct? Unfortunately I don't have a Windows/MSVC15 setup at hand to test.

Reply:
Commented in https://github.com/j-rivero/fcl/commit/e509a3d1659591bb4bacc98638ff74e505c5fee7 .

Reply:
Sorry, I actually commented in https://github.com/flexible-collision-library/fcl/commit/e509a3d1659591bb4bacc98638ff74e505c5fee7 .